File: app/src/at-nhatnguyen/java/asiantech/internship/summer/javacore/Bai10.java
package asiantech.internship.summer.javacore;

import java.util.Scanner;

public class Bai10 {

    private static int input() {
        Scanner scan = new Scanner(System.in);
        boolean check = false;
        int n = 0;
        while (!check) {
            System.out.print(" ");
            try {
                n = scan.nextInt();
                check = true;
            } catch (Exception e) {
                System.out.println("Ban phai nhap so :");
                scan.nextLine();
            }
        }
        return (n);
    }

    private static void result(int[] a, int k) {
        int i;
        System.out.println();
        for (i = 1; i <= k; i++) {
            System.out.print(" " + a[i]);
        }
    }

    private static void try_backTrack(int[] a, int n, int k, int i) {
        int j;
        for (j = a[i - 1] + 1; j <= (n - k + i); j++) {
            a[i] = j;
            if (i == k) result(a, k);
            else try_backTrack(a, n, k, i + 1);
        }
    }

    public static void main(String[] args) {
        System.out.println("Nhap n");
        int n = input();
        int[] array = new int[n + 1];
        int k;
        System.out.println("Liet ke tat ca cac tap con k phan tu cua 1,2,..," + n + " : ");
        for (k = 1; k <= n; k++) {
            System.out.println("\n Tap con " + k + " phan tu: ");
            try_backTrack(array, n, k, 1);
        }
    }
}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai10.kt
package asiantech.internship.winter.javacore

import java.util.*


/*
Bài 10. Viết chương trình liệt kê tất cả các tập con k phần tử của 1, 2, ..,n (k≤n).
 */
object Bai10 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter n: ")
        val n = scanner.nextInt()

        val array = IntArray(n + 1)

        var k = 1
        while (k <= n) {
            println("\n Subsets $k-element: ")
            tryBacktrack(array, n, k, 1)
            k++
        }
    }

    private fun result(a: IntArray, k: Int) {
        var i = 1
        println()
        while (i <= k) {
            print("  " + a[i])
            i++
        }
    }

    private fun tryBacktrack(a: IntArray, n: Int, k: Int, i: Int) {
        var j: Int = a[i - 1] + 1
        while (j <= n - k + i) {
            a[i] = j
            if (i == k)
                result(a, k)
            else
                tryBacktrack(a, n, k, i + 1)
            j++
        }
    }
}